extensions/wikia/Wall/WallTabsRenderer.php

```diff
@@ -0,0 +1,104 @@
+<?php
+/**
+ * Copyright (C) 2017 Wikia, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ * https://www.gnu.org/copyleft/gpl.html
+ *
+ * @file
+ */
+
+class WallTabsRenderer extends ContextSource {
+	public function __construct( IContextSource $context ) {
+		$this->setContext( $context );
+	}
+
+	public function renderUserPageContentActions( array &$contentActions ) {
+		if ( empty( $contentActions['namespaces'] ) || empty( $contentActions['namespaces']['user_talk'] ) ) {
+			return;
+		}
+
+		$userWallTitle = $this->getTitle()->getTalkPage();
+
+		$contentActions['namespaces']['user_talk']['text'] = $this->msg( 'wall-message-wall' )->text();
+		$contentActions['namespaces']['user_talk']['href'] = $userWallTitle->getLocalUrl();
+
+		// BugId:23000 Remove the class="new" to prevent the link from being displayed as a redlink in monobook.
+		$tabClasses = $contentActions['namespaces']['user_talk']['class'];
+		$contentActions['namespaces']['user_talk']['class'] =
+			array_filter( $tabClasses, function ( $class ) {
+				return $class !== 'new';
+			} );
+	}
+
+	public function renderWallContentActions( array &$contentActions ) {
+		$messageTitle = $this->getMessageTitle();
+		$wallTitleText = $messageTitle->getBaseText();
+
+		$owningUser = User::newFromName( $wallTitleText, $validateUserName = false );
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused "$validateUserName" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)